src/ngraph/runtime/cpu/pass/rnn_fusion.cpp

```
@@ -0,0 +1,566 @@
+
+        NGRAPH_DEBUG << " In recurrent RNN fusion callback";
+
+        auto ht_1_label = m.get_bound_nodes_for_pattern(rpattern_ht_1);
+
+        // determine the ht and xt
+        std::shared_ptr<ngraph::Node> src_layer = nullptr;
+        std::shared_ptr<ngraph::Node> src_iter = nullptr;
+
+        auto xt_node_array = m.get_bound_nodes_for_pattern(xt);
+        auto hidden_ht_array = m.get_bound_nodes_for_pattern(rpattern_ht_1);
+
+        // since we dont have metadata to differentiate between xt and ht_1
+        // we will be using the broadcasted constant initilization of the first LSTM cell
+        // in the RNN layer to identify ht_1
+        if (std::dynamic_pointer_cast<op::Broadcast>(xt_node_array[xt_node_array.size() - 1]) &&
+            std::dynamic_pointer_cast<op::Constant>(
+                xt_node_array[xt_node_array.size() - 1]->get_argument(0)))
+        {
+            std::vector<std::shared_ptr<pattern::op::Label>> src_layer_labels{rpattern_ht_1};
+            src_layer = compute_rnn_args(src_layer_labels, m, true);
+
+            std::vector<std::shared_ptr<pattern::op::Label>> src_iter_labels{xt, ct_1};
+            src_iter = compute_rnn_args(src_iter_labels, m);
+        }
+        else if (std::dynamic_pointer_cast<op::Broadcast>(
+                     hidden_ht_array[hidden_ht_array.size() - 1]) &&
+                 std::dynamic_pointer_cast<op::Constant>(
+                     hidden_ht_array[hidden_ht_array.size() - 1]->get_argument(0)))
+        {
+            std::vector<std::shared_ptr<pattern::op::Label>> src_layer_labels{xt};
+            src_layer = compute_rnn_args(src_layer_labels, m, true);
+
+            std::vector<std::shared_ptr<pattern::op::Label>> src_iter_labels{rpattern_ht_1, ct_1};
+            src_iter = compute_rnn_args(src_iter_labels, m);
+        }
+        else
+        {
+            // dont fuse, if the PM dint discover all the cell's belonging to RNN layer.
+            return false;
```

throw exception instead of returning